rocketpy/rocket/aero_surface/generic_surface.py

```diff
@@ -0,0 +1,444 @@
+        stream_mach : float
+            The Mach number of the airflow.
+        rho : float
+            Air density.
+        cp : Vector
+            Center of pressure coordinates in the body frame.
+        omega: tuple[float, float, float]
+            Tuple containing angular velocities around the x, y, z axes.
+        reynolds : float
+            Reynolds number.
+        omega: tuple of float
+            Tuple containing angular velocities around the x, y, z axes.
+
+        Returns
+        -------
+        tuple of float
+            The aerodynamic forces (lift, side_force, drag) and moments
+            (pitch, yaw, roll) in the body frame.
+        """
+        # Stream velocity in standard aerodynamic frame
+        stream_velocity = -stream_velocity
+
+        # Angles of attack and sideslip
+        alpha = np.arctan2(stream_velocity[1], stream_velocity[2])
+        beta = np.arctan2(-stream_velocity[0], stream_velocity[2])
+
+        # Compute aerodynamic forces and moments
+        lift, side, drag, pitch, yaw, roll = self._compute_from_coefficients(
+            rho,
+            stream_speed,
+            alpha,
+            beta,
+            stream_mach,
+            reynolds,
+            omega[0],
+            omega[1],
+            omega[2],
+        )
+
+        # Conversion from aerodynamic frame to body frame
+        rotation_matrix = Matrix(
+            [
+                [1, 0, 0],
+                [0, math.cos(alpha), -math.sin(alpha)],
+                [0, math.sin(alpha), math.cos(alpha)],
+            ]
+        ) @ Matrix(
+            [
+                [math.cos(beta), 0, -math.sin(beta)],
+                [0, 1, 0],
+                [math.sin(beta), 0, math.cos(beta)],
+            ]
+        )
+        R1, R2, R3 = rotation_matrix @ Vector([side, -lift, -drag])
+
+        # Dislocation of the aerodynamic application point to CDM
+        M1, M2, M3 = Vector([pitch, yaw, roll]) + (cp ^ Vector([R1, R2, R3]))
+
+        return R1, R2, R3, M1, M2, M3
+
+    def _process_input(self, input_data, coeff_name):
+        """Process the input data, either as a CSV file or a callable function.
+
+        Parameters
+        ----------
+        input_data : str or callable
+            Input data to be processed, either a path to a CSV or a callable.
+        coeff_name : str
+            Name of the coefficient being processed for error reporting.
+
+        Returns
+        -------
+        Function
+            Function object with 4 input arguments (alpha, beta, mach, reynolds).
+        """
+        if isinstance(input_data, str):
+            # Input is assumed to be a file path to a CSV
+            return self.__load_csv(input_data, coeff_name)
+        elif isinstance(input_data, Function):
+            if input_data.__dom_dim__ != 7:
+                raise ValueError(
+                    f"{coeff_name} function must have 7 input arguments"
+                    " (alpha, beta, mach, reynolds)."
```

Comment:
I think you forgot the last three arguments 
```suggestion
                    f"{coeff_name} function must have 7 input arguments"
                    " (alpha, beta, mach, reynolds, pitch_rate, yaw_rate, roll_rate)."
```